dotnet-authserver/src/TeacherIdentity.AuthServer/AuthenticationState.cs

```diff
@@ -678,6 +670,26 @@ private bool ShouldRedirectToLandingPage()
 
         return !OAuthState.HasAnyScope(ignoredScopes) && !UserId.HasValue;
     }
+
+    private void UpdateAuthenticationStateWithUserDetails(User user)
+    {
+        UserId = user.UserId;
+        EmailAddress = user.EmailAddress;
+        MobileNumber = user.MobileNumber;
```

Comment:
Added updates to MobileNumber and EmailAddress here. There's a debug assertion in OnEmailAddressVerified that indicates we may have different cases, but assuming that's ok.
`Debug.Assert(user.EmailAddress.Equals(EmailAddress, StringComparison.OrdinalIgnoreCase));`